backend/app/data_processing/gesture.py

```diff
@@ -0,0 +1,101 @@
+import math
+from app.synthesis.synthesizers import generate_sine_wave_with_envelope
+import pyaudio
+
+# factor by which we compress coordinates
+FACTOR = 10
+
+
+def compress_coordinates(gestures):
+    """
+    :param gestures: list of list of coordinates (each coordinate a dictionary) received from user
+    input via
+    frontend
+    :return: (in the same format as input), shorter list of coordinates, each representing a
+    compressed coordinate, compressed by taking averages of groups of coordinates
+    """
+    result = []
+    for gesture in gestures:
+        current_result = []
+        for i in range(0, len(gesture)-FACTOR, FACTOR):
+            compressed_coord = {
```

Comment:
It's helpful for the reader to unpack this kind of thing a bit (separating the dictionary assignment from the calculation, and naming the intermediary):

```
avg_x = (gesture[i]["x"] + gesture[i+FACTOR-1]["x"]) / 2
avg_y = (gesture[i]["y"] + gesture[i+FACTOR-1]["y"]) / 2
compressed_coord = {
    "x": avg_x, 
    "y": avg_y,
}
```

Reply:
Also, the docs say we're averaging groups of coordinates, but we're actually just taking the mean of the first and last coords of a sample. It would be pretty straightforward to actually take the mean of the whole group in an inner loop. Or just update the docs so it's a little clearer what's going on, since there probably isn't much of a win in making this fully correct.